File: go/dnse/api_auth.go
package dnse

// CreateTradingToken generates a Trading Token required for order placement.
func (c *Client) CreateTradingToken(otpType, passcode string, dryRun bool) (int, []byte, error) {
	body := map[string]string{
		"otpType":  otpType,
		"passcode": passcode,
	}
	return c.Request("POST", "/registration/trading-token", nil, body, nil, dryRun)
}

// SendEmailOTP requests an OTP sent to your registered email.
func (c *Client) SendEmailOTP(dryRun bool) (int, []byte, error) {
	return c.Request("POST", "/registration/send-email-otp", nil, nil, nil, dryRun)
}

File: rust/src/rest/api_auth.rs
use super::client::DnseClient;
use reqwest::Method;
use serde_json::json;

impl DnseClient {
    pub async fn create_trading_token(&self, otp_type: &str, passcode: &str, dry_run: bool) -> Result<(u16, Vec<u8>), Box<dyn std::error::Error>> {
        let body = json!({
            "otpType": otp_type,
            "passcode": passcode,
        });
        self.request(Method::POST, "/registration/trading-token", None, Some(&body), None, dry_run).await
    }

    pub async fn send_email_otp(&self, dry_run: bool) -> Result<(u16, Vec<u8>), Box<dyn std::error::Error>> {
        self.request(Method::POST, "/registration/send-email-otp", None, None, None, dry_run).await
    }
}
